activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java

```diff
@@ -831,10 +832,8 @@ public void stop() throws Exception {
             this.scheduler.stop();
             this.scheduler = null;
         }
-        if (services != null) {
-            for (Service service : services) {
-                stopper.stop(service);
-            }
+        for (Service service : services) {
```

Comment:
I think we should still check if `services` is not `null` as it can be passed programmatically.

Reply:
Interesting, because services is marked as final when it's instantiated in line 180. So services should never be null. Am I read the code correctly?